pkg/controller/actions/deploy/action_deploy.go

```diff
@@ -214,42 +291,32 @@ func (a *Action) deploy(
 
 		var err error
 
+		ops := []client.PatchOption{
+			client.ForceOwnership,
+			client.FieldOwner(fo),
+		}
+
 		switch a.deployMode {
 		case ModePatch:
-			deployedObj, err = a.patch(
-				ctx,
-				rr.Client,
-				&obj,
-				current,
-				client.ForceOwnership,
-				client.FieldOwner(fo))
+			deployedObj, err = a.patch(ctx, rr.Client, &obj, current, ops...)
 		case ModeSSA:
-			deployedObj, err = a.apply(
-				ctx,
-				rr.Client,
-				&obj,
-				current,
-				client.ForceOwnership,
-				client.FieldOwner(fo))
+			deployedObj, err = a.apply(ctx, rr.Client, &obj, current, ops...)
 		default:
 			err = fmt.Errorf("unsupported deploy mode %s", a.deployMode)
 		}
 
 		if err != nil {
-			return false, err
+			return false, client.IgnoreNotFound(err)
 		}
 
 		deployed = true
```

Comment:
why we need  `var deployed bool`  in deploy()
in case resources.GetAnnotation(&obj, annotations.ManagedByODHOperator) == "false":
it did not set value, so will "return false, nil"
in deployCRD() , will always "return true, nil" if no error

Reply:
done